fe/fe-core/src/main/java/com/starrocks/catalog/HiveTable.java

```diff
@@ -590,14 +590,14 @@ public void readFields(DataInput in) throws IOException {
 
     @Override
     public void onCreate() {
-        if (this.resourceName != null) {
-            GlobalStateMgr.getCurrentState().getMetastoreEventsProcessor().registerTable(this);
-        }
+        hiveRepository.getCounter().add(resourceName, hiveDbName, hiveTableName);
+        GlobalStateMgr.getCurrentState().getMetastoreEventsProcessor().registerTable(this);
     }
 
     @Override
     public void onDrop() {
-        if (this.resourceName != null) {
+        hiveRepository.getCounter().reduce(resourceName, hiveDbName, hiveTableName);
+        if (hiveRepository.getCounter().get(resourceName, hiveDbName, hiveTableName) == 0) {
```

Comment:
The `reduce and get` should be combined into a atomic action, otherwise there would two threads contend to clearCache

Reply:
It was written like this at first, but later modified wrong. How stupid am I. thanks~